internal/webhook/webhook.go

```diff
@@ -0,0 +1,193 @@
+var _ webhook.CustomDefaulter = (*Handler)(nil)
+var _ webhook.CustomValidator = (*Handler)(nil)
+
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type
+func (Handler) Default(_ context.Context, obj runtime.Object) error {
+	oidc, ok := obj.(*authenticationv1alpha1.OpenIDConnect)
+	if !ok {
+		return fmt.Errorf("expected *authenticationv1alpha1.OpenIDConnect but got %T", obj)
+	}
+
+	log.Info("Defaulting OpenIDConnect resource", "name", oidc.Name)
+
+	var (
+		defaultUsernameClaim = "sub"
+		defaultGroupsClaim   = "groups"
+	)
+
+	if oidc.Spec.UsernameClaim == nil {
+		oidc.Spec.UsernameClaim = &defaultUsernameClaim
+	}
+
+	if oidc.Spec.GroupsClaim == nil {
+		oidc.Spec.GroupsClaim = &defaultGroupsClaim
+	}
+
+	if len(oidc.Spec.SupportedSigningAlgs) == 0 {
+		oidc.Spec.SupportedSigningAlgs = []authenticationv1alpha1.SigningAlgorithm{authenticationv1alpha1.RS256}
+	}
+
+	return nil
+}
+
+// ValidateCreate validates the object on creation.
+// The optional warnings will be added to the response as warning messages.
+// Return an error if the object is invalid.
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type
+func (Handler) ValidateCreate(_ context.Context, obj runtime.Object) (admission.Warnings, error) {
+	oidc, ok := obj.(*authenticationv1alpha1.OpenIDConnect)
+	if !ok {
+		return nil, fmt.Errorf("expected *authenticationv1alpha1.OpenIDConnect but got %T", obj)
+	}
+
+	log.Info("Validating OpenIDConnect resource", "operation", "create", "name", oidc.Name)
+
+	return nil, validate(oidc).ToAggregate()
+}
+
+// ValidateUpdate validates the object on update.
+// The optional warnings will be added to the response as warning messages.
+// Return an error if the object is invalid.
+// ValidateUpdate implements webhook.CustomValidator so a webhook will be registered for the type
+func (Handler) ValidateUpdate(_ context.Context, _, obj runtime.Object) (admission.Warnings, error) {
+	oidc, ok := obj.(*authenticationv1alpha1.OpenIDConnect)
+	if !ok {
+		return nil, fmt.Errorf("expected *authenticationv1alpha1.OpenIDConnect but got %T", obj)
+	}
+
+	log.Info("Validating OpenIDConnect resource", "operation", "update", "name", oidc.Name)
+
+	return nil, validate(oidc).ToAggregate()
+}
+
+// ValidateDelete validates the object on deletion.
+// The optional warnings will be added to the response as warning messages.
+// Return an error if the object is invalid.
+// ValidateDelete implements webhook.CustomValidator so a webhook will be registered for the type
+func (Handler) ValidateDelete(_ context.Context, obj runtime.Object) (admission.Warnings, error) {
+	oidc, ok := obj.(*authenticationv1alpha1.OpenIDConnect)
+	if !ok {
+		return nil, fmt.Errorf("expected *authenticationv1alpha1.OpenIDConnect but got %T", obj)
+	}
+
+	log.Info("Validating OpenIDConnect resource", "operation", "delete", "name", oidc.Name)
```

Comment:
I think that we can remove the word "resource" as it does not bring much info. Same for the other occurrences.
```suggestion
	log.Info("Validating OpenIDConnect", "operation", "delete", "name", oidc.Name)
```